src/IK/src/GravityTask.cpp

```diff
@@ -163,46 +147,35 @@ bool GravityTask::update()
 
     m_isValid = false;
 
-    // Normalize Accelerometer vector
-    m_accDenomNorm = sqrt(pow(m_currentAcc(0),2) + pow(m_currentAcc(1),2) + pow(m_currentAcc(2),2));
-
-    for (size_t i = 0; i < m_currentAcc.size(); i++)
-    {
-        m_currentAccNorm(i) = m_currentAcc(i) / m_accDenomNorm;
-    }
+    auto targetRotation = toEigen(m_kinDyn->getWorldTransform(m_targetFrameIndex).getRotation());
 
-    m_currentAccNorm = toEigen(m_kinDyn->getWorldTransform(m_targetFrameName).getRotation()) * m_currentAccNorm;
+    Eigen::Vector3d desiredZDirectionAbsolute = targetRotation * m_desiredZDirectionBody;
+    Eigen::Vector3d feedforwardAbsolute = targetRotation * m_feedForwardBody;
 
-    // get the relative jacobian
-    if (!m_kinDyn->getRelativeJacobian(m_baseIndex, m_targetFrameIndex, m_relativeJacobian))
+    if (!m_kinDyn->getFrameFreeFloatingJacobian(m_targetFrameIndex, m_jacobian))
     {
-        log()->error("[GravityTask::update] Unable to get the relative jacobian.");
+        log()->error("[GravityTask::update] Unable to get the frame jacobian.");
         return m_isValid;
     }
 
-    // get the angular part of the jacobian
-    m_Angularjacobian.bottomRightCorner(3, m_kinDyn->getNrOfDegreesOfFreedom()) = m_relativeJacobian.bottomRightCorner(3, m_kinDyn->getNrOfDegreesOfFreedom());
+    m_A = m_Am * m_jacobian.bottomRows<3>();
+    m_b << m_kp * m_bm * desiredZDirectionAbsolute + feedforwardAbsolute.topRows<2>();
 
-    m_A = m_Am * m_Angularjacobian;
-    m_b << ( m_kp * m_bm * m_currentAccNorm) + 1000 * m_currentGyro.block<2,1>(1,0);
-
-    // A and b are now valid
     m_isValid = true;
     return m_isValid;
 }
 
-bool GravityTask::setEstimateGravityDir(const Eigen::Ref<const Eigen::VectorXd> currentGravityDir)
+void GravityTask::setDesiredGravityDirectionInTargetFrame(
+    const Eigen::Ref<const Eigen::Vector3d> desiredGravityDirection)
 {
-    m_currentAcc = currentGravityDir;
-
-    return true;
+    m_desiredZDirectionBody = desiredGravityDirection;
+    m_desiredZDirectionBody.normalize();
 }
 
-bool GravityTask::setGyroscope(const Eigen::Ref<const Eigen::VectorXd> currentGyro)
+void GravityTask::setFeedForwardVelocityInTargetFrame(
+    const Eigen::Ref<const Eigen::Vector3d> feedforwardVelocity)
 {
-    m_currentGyro = currentGyro * (M_PI)/180;
```

Comment:
The Gyro reading of the IMUs of the WEART is in [deg/s]. That is why here I converted to radians. Unless we have to specify this method accepts only radians.

[raw_data_touchDIVER.pdf](https://github.com/ami-iit/bipedal-locomotion-framework/files/12313917/raw_data_touchDIVER.pdf)

Reply:
RIght. I should add in the documentation that the input is expected to be in radians.